File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_sum = 0
        max_sum = float('-inf')
        for n in nums:
            current_sum += n
            max_sum = max(max_sum, current_sum)
            if current_sum < 0:
                current_sum = 0
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int bestSum = 0;
        int ans = Integer.MIN_VALUE;
        for(int i = 0; i<nums.length; i++){
            int choice1 = nums[i]+bestSum;
            int choice2 = nums[i];
            bestSum = Math.max(choice1, choice2);
            ans = Math.max(ans,bestSum);
        }
        return ans;
    }
}